lib/rspec/core/memoized_helpers.rb

```diff
@@ -285,6 +288,9 @@ def let(name, &block)
           # We have to pass the block directly to `define_method` to
           # allow it to use method constructs like `super` and `return`.
           raise "#let or #subject called without a block" if block.nil?
+          raise(
+            "#let or #subject called with a reserved name ##{name}"
+          ) if ReservedKeywords.include? name.to_s
```

Comment:
Actually, since we only have `initialize` for now, I think it's simpler and more efficient to remove the constant and just check `if name == :initialize` here.  Declaring the constant and making it general feels like a violation of YAGNI given we don't have more than one method to check.